R/validate_inputs.R

```diff
@@ -90,3 +90,21 @@ do_validate_anc <- function(anc) {
     c("iso3", "area_id", "period", "sex", "age_group_id", "indicator", "value"))
   anc
 }
+
+#' Validate survey data file.
+#'
+#' Check that survey data file can be read and return serialised data.
+#'
+#' @param anc Path to input survey file.
+#'
+#' @return An error if invalid.
+#' @keywords internal
+do_validate_survey <- function(survey) {
+  survey <- read_csv(survey, header = TRUE)
+  assert_single_country(survey, "anc")
```

Comment:
"anc" should be "survey"

Reply:
Not sure which level this makes most sense at but we should be testing the unhappy path as well - i.e. testing that bad survey data fails the validation. I can't see that happening anywhere? Such a test would catch mistakes like this that only pertain to the error message returned!

Reply:
Fixed. Yeah I should have added that. There was one in for the programme but I had forgotten for anc and survey data. I've added a test for each now - though they only check 1 of the failure conditions. I think probably I need to add some helper code to setup the malformed test data to avoid having a proliferation of test data within the package otherwise we're going to end up with 1 file for every possible type of error for each input type.